### PR TITLE
Ssl-gateway

### PR DESCRIPTION
**Pull request by yuval-k**
This PR changes how a gateway works in respect to ssl:
A gateway has a ssl boolean now, which detemines if it is an ssl gateway or not.
an (ssl\no-ssl) gateway will only pick up (ssl\no-ssl) virtual services.

next inline:
adding command line to add ssl to virtualservices
fix the doc here:
https://github.com/solo-io/solo-docs/pull/31

### CODE REVIEW

File: install/helm/gloo/values-gateway-template.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 namespace:
-  create: false
+  create: true
```
Comment (ilackarms) [inline install/helm/gloo/values-gateway-template.yaml]: why are we now switching default value here to true?

Comment (yuval-k) [reply install/helm/gloo/values-gateway-template.yaml]: git merge mess :/

**Comment by yuval-k** [comment]
note: ran gateway kube e2e tests and they pass

**Pull request merged**